### PR TITLE
duress password ui

### PR DESCRIPTION
**Pull request by empratyush**

### CODE REVIEW

File: src/com/android/settings/passwordDuress/ChooseDuressPinOrPassword.java [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+
+            int type = LockPatternUtils.CREDENTIAL_TYPE_NONE;
+            if (ScreenLockType.PIN.preferenceKey.equals(preference.getKey())){
+                type =  DevicePolicyManager.PASSWORD_QUALITY_NUMERIC;
+            }
+            if (ScreenLockType.PASSWORD.preferenceKey.equals(preference.getKey())){
+                type = DevicePolicyManager.PASSWORD_QUALITY_ALPHANUMERIC;
+            }
+
+            if (ScreenLockType.NONE.preferenceKey.equals(preference.getKey())){
```
Comment (flawedworld) [inline src/com/android/settings/passwordDuress/ChooseDuressPinOrPassword.java:141]: I don't see the point of having none as an option in this UI.

File: res/values/strings.xml [hunk 2]
```diff
@@ -13748,4 +13756,17 @@
 
     <string name="battery_share_description">Charge other devices by placing them on the back of your phone</string>
     <string name="battery_share">Battery share</string>
+
+    <string name="choose_duress_password_title">Choose a new Duress Lock</string>
+    <string name="update_duress_password_title">Update existing Duress Lock</string>
+
+    <string name="duress_pin_exist">PIN</string>
+    <string name="duress_password_exist">PASSWORD</string>
```
Comment (flawedworld) [inline res/values/strings.xml]: Make this lowercase

Comment (flawedworld) [reply res/values/strings.xml]: "Password"

File: res/values/strings.xml [hunk 3]
```diff
@@ -1502,6 +1502,7 @@
     <!--  Main Security lock settings --><skip />
     <!--  Title for PreferenceScreen to launch picker for security method when there is none [CHAR LIMIT=22] -->
     <string name="unlock_set_unlock_launch_picker_title">Screen lock</string>
+    <string name="duress_password_title">Duress Password</string>
```
Comment (jorgeluiscarrillo) [inline res/values/strings.xml]: Lowercase `password` for consistency with the other settings

File: res/values/strings.xml [hunk 4]
```diff
@@ -13748,4 +13756,19 @@
 
     <string name="battery_share_description">Charge other devices by placing them on the back of your phone</string>
     <string name="battery_share">Battery share</string>
+
+    <string name="choose_duress_password_title">Choose a new Duress Lock</string>
+    <string name="update_duress_password_title">Update existing Duress Lock</string>
```
Comment (jorgeluiscarrillo) [inline res/values/strings.xml]: Lowercase `duress lock` for consistency

File: res/values/strings.xml [hunk 5]
```diff
@@ -13748,4 +13756,19 @@
...
+
+    <string name="duress_pin_exist">PIN</string>
+    <string name="duress_password_exist">PASSWORD</string>
+    <string name="duress_pin_and_password_exist">PIN &amp; PASSWORD</string>
+    <string name="added_a_duress_pin_and_password">NONE</string>
```
Comment (jorgeluiscarrillo) [inline res/values/strings.xml]: Similar to screen lock setting, don't use all caps for `Password` and `None`

File: res/values/strings.xml [hunk 6]
```diff
@@ -4425,6 +4426,12 @@
     <!-- About phone settings screen, Safety Legal dialog title until the link is fully loaded -->
     <string name="settings_safetylegal_activity_loading">Loading\u2026</string>
 
+    <string name="duress_password_choose_your_password_header">Set a duress password</string>
+    <string name="duress_password_choose_your_pin_header">Set a duress PIN</string>
+
+    <string name="duress_password_choose_password_description">Set a password to wipe device without unlocking</string>
+    <string name="duress_password_choose_pin_description">Set a pin to wipe device without unlocking</string>
```
Comment (jorgeluiscarrillo) [inline res/values/strings.xml]: Fix: `wipe device` should be `wipe the device`

File: res/values/strings.xml [hunk 7]
```diff
@@ -13748,4 +13756,19 @@
...
+    <string name="duress_password_exist">PASSWORD</string>
+    <string name="duress_pin_and_password_exist">PIN &amp; PASSWORD</string>
+    <string name="added_a_duress_pin_and_password">NONE</string>
+
+    <string name="duress_descriptions">Duress mode in GrapheneOS will only work on if you are using a lockscreen which is the same as what you have configured for duress. Duress mode will wipe the device fully, including user data and eSIMS. Duress mode does not erase the presence of GrapheneOS of the device. After duress mode has been triggered by the user, the device will act as if it is a fresh installation of GrapheneOS.</string>
+    <string name="duress_title">Add duress password?</string>
+    <string name="delete_duress_title">Delete duress password?</string>
+    <string name="delete_duress_descriptions">Duress password in GrapheneOS allow you to quickly wipe your device without unlocking the device.</string>
+
+    <string name="add_a_lock_screen_first">Add a screen lock first</string>
```
Comment (jorgeluiscarrillo) [inline res/values/strings.xml]: I would change this to `Requires a screen lock`

File: res/values/strings.xml [hunk 8]
```diff
@@ -13748,4 +13756,19 @@
...
+
+    <string name="duress_pin_exist">PIN</string>
+    <string name="duress_password_exist">PASSWORD</string>
+    <string name="duress_pin_and_password_exist">PIN &amp; PASSWORD</string>
+    <string name="added_a_duress_pin_and_password">NONE</string>
+
+    <string name="duress_descriptions">Duress mode in GrapheneOS will only work on if you are using a lockscreen which is the same as what you have configured for duress. Duress mode will wipe the device fully, including user data and eSIMS. Duress mode does not erase the presence of GrapheneOS of the device. After duress mode has been triggered by the user, the device will act as if it is a fresh installation of GrapheneOS.</string>
+    <string name="duress_title">Add duress password?</string>
+    <string name="delete_duress_title">Delete duress password?</string>
+    <string name="delete_duress_descriptions">Duress password in GrapheneOS allow you to quickly wipe your device without unlocking the device.</string>
```
Comment (jorgeluiscarrillo) [inline res/values/strings.xml]: Fix: `allow` should be `allows`
Consider removing `in GrapheneOS` as it seems unnecessary
Change `wipe your device without unlocking the device.` to `wipe your device without unlocking it.`

File: res/values/strings.xml [hunk 9]
```diff
@@ -13748,4 +13756,19 @@
...
+
+    <string name="choose_duress_password_title">Choose a new Duress Lock</string>
+    <string name="update_duress_password_title">Update existing Duress Lock</string>
+
+    <string name="duress_pin_exist">PIN</string>
+    <string name="duress_password_exist">PASSWORD</string>
+    <string name="duress_pin_and_password_exist">PIN &amp; PASSWORD</string>
+    <string name="added_a_duress_pin_and_password">NONE</string>
+
+    <string name="duress_descriptions">Duress mode in GrapheneOS will only work on if you are using a lockscreen which is the same as what you have configured for duress. Duress mode will wipe the device fully, including user data and eSIMS. Duress mode does not erase the presence of GrapheneOS of the device. After duress mode has been triggered by the user, the device will act as if it is a fresh installation of GrapheneOS.</string>
```
Comment (jorgeluiscarrillo) [inline res/values/strings.xml]: Fix: Remove `on` from `will only work on if`
Fix: `eSIMS` should be `eSIMs`
Fix: `GrapheneOS of the device` should be `GrapheneOS on the device`
Change `lockscreen which is the same` to `screen lock that matches`
Change `device will act` to `device will function`

The wording for some things can also be improved but I am not sure what info is / isn't important to show here.

**Review by jorgeluiscarrillo — COMMENTED** [review]
Just suggesting some minor string fixes / improvements

File: res/values/strings.xml (hunk 5 above)
Comment (empratyush) [reply res/values/strings.xml]: why ?

Comment (jorgeluiscarrillo) [reply res/values/strings.xml]: Just to be consistent with the screen lock setting. Unless I am mistaken where this is shown

Comment (empratyush) [reply res/values/strings.xml]: it's already  [consistent ](https://imgur.com/a/OlczGFu)with the screen lock setting.

Comment (jorgeluiscarrillo) [reply res/values/strings.xml]: PIN is ok.
I am referring to PASSWORD and NONE. The screen lock setting uses `Password` or `None` when set, so this change would make it consistent.
`PIN & PASSWORD` would be `PIN & Password`

Comment (empratyush) [reply res/values/strings.xml]: ah i see, thanks.

File: res/values/duress_string.xml [hunk 10]
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <string name="duress_password_title">Duress password</string>
+
+    <string name="duress_password_choose_your_password_header">Set a duress password</string>
+    <string name="duress_password_choose_your_pin_header">Set a duress PIN</string>
+
+    <string name="duress_password_choose_password_description">Set a password to wipe device without unlocking</string>
+    <string name="duress_password_choose_pin_description">Set a pin to the wipe device without unlocking</string>
```
Comment (matchboxbananasynergy) [inline res/values/duress_string.xml]: Shouldn't this say "PIN" instead of "pin"

File: res/values/duress_string.xml [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+
+    <string name="keywords_duress_password">Duress Pin, Duress Password</string>
```
Comment (matchboxbananasynergy) [inline res/values/duress_string.xml]: "PIN" instead of "Pin"

File: res/values/duress_string.xml [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+
+    <string name="choose_duress_password_title">Choose a new duress lock</string>
```
Comment (matchboxbananasynergy) [inline res/values/duress_string.xml]: "duress lock" here is not capitalizing, but it is capitalized on the next line. Which of the two is correct here?

File: res/values/duress_string.xml (hunk 10 above)
Comment (empratyush) [reply res/values/duress_string.xml]: fixed.

**Comment by thestinger** [comment]
@empratyush You need to change the base branch to `14` for these and other PRs. Did that for this one.

File: src/com/android/settings/passwordDuress/DuressPasswordUi.kt [hunk 13]
```diff
@@ -0,0 +1,276 @@
...
+        }
+
+        saveAndFinish(pin = pin, password = password)
+    }
+
+    override fun onCreate(savedState: Bundle?) {
+        super.onCreate(savedState)
+        setTheme(com.google.android.material.R.style.Theme_Material3_DayNight_NoActionBar)
+        com.google.android.material.color.DynamicColors.applyToActivityIfAvailable(this)
+        window.navigationBarColor = android.graphics.Color.TRANSPARENT
```
Comment (quh4gko8) [inline src/com/android/settings/passwordDuress/DuressPasswordUi.kt:183]: Status bar theming has low contrast in dark mode with the text in it. Using transparent color with the status bar color, along with appropriate theming flags would look better, with change something like https://github.com/repository-staging/platform_packages_apps_Settings/commit/963f7b338da2e00deaf16a0296005ad42b499606

**Comment by empratyush** [comment]
replaced by newer implementation.

**Pull request closed without merging**